src/main/java/org/opensearch/timeseries/ml/ModelColdStart.java

```diff
@@ -525,12 +527,17 @@ public static <T extends ThresholdedRandomCutForest.Builder<T>> T applyImputatio
                 case ZERO:
                     return builder.imputationMethod(ImputationMethod.ZERO);
                 case FIXED_VALUES:
-                    // we did validate default fill is not empty and size matches enabled feature number in Config's constructor
-                    return builder.imputationMethod(ImputationMethod.FIXED_VALUES).fillValues(imputationOption.getDefaultFill().get());
+                    // we did validate default fill is not empty, size matches enabled feature number in Config's constructor,
+                    // and feature names matches existing features
+                    List<String> enabledFeatureName = config.getEnabledFeatureNames();
+                    double[] fillValues = new double[enabledFeatureName.size()];
+                    Map<String, Double> defaultFillMap = imputationOption.getDefaultFill();
+                    for (int i = 0; i < enabledFeatureName.size(); i++) {
+                        fillValues[i] = defaultFillMap.get(enabledFeatureName.get(i));
```

Comment:
could defaultFillMap.get(enabledFeatureName.get(i)) return a NPE if user updates which features are enabled? 
Might be not needed but we can add null check or change to `getOrDefault` if we think its a possibility

Reply:
When there is an update, we will require stop detectors and we redo the feature name matching. So null check is not needed. Also, I don't know which default we should provide here if that ever happens.